crates/figma_import/src/figma_schema.rs

```diff
@@ -1070,3 +1080,164 @@ pub struct ProjectFilesResponse {
+    pub default_mode_id: String,
+    pub id: String,
+    pub name: String,
+    pub remote: bool,
+    pub modes: Vec<ModeBinding>,
+    pub key: String,
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+pub enum VariableType {
+    Boolean,
+    Float,
+    String,
+    Color,
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct VariableAlias {
+    pub r#type: String,
+    pub id: String,
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone, PartialEq)]
+#[serde(untagged)]
+pub enum VariableAliasOrList {
+    Alias(VariableAlias),
+    List(Vec<VariableAlias>),
+}
+impl VariableAliasOrList {
+    fn get_name(&self) -> Option<String> {
+        match self {
+            VariableAliasOrList::Alias(alias) => {
+                return Some(alias.id.clone());
+            }
+            VariableAliasOrList::List(list) => {
+                let alias = list.first();
+                if let Some(alias) = alias {
+                    return Some(alias.id.clone());
+                }
+            }
+        }
+        None
+    }
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct BoundVariables {
+    #[serde(flatten)]
+    variables: HashMap<String, VariableAliasOrList>,
+}
+impl BoundVariables {
+    pub(crate) fn has_var(&self, var_name: &str) -> bool {
+        self.variables.contains_key(var_name)
+    }
+
+    pub(crate) fn get_variable(&self, var_name: &str) -> Option<String> {
+        let var = self.variables.get(var_name);
+        if let Some(var) = var {
+            return var.get_name();
+        }
+        None
+    }
+}
+
+// We use the "untagged" serde attribute because the value of a variable is
+// described in a hash where we don't know the format of the value based on the
+// key. The value format is different depending on the type of variable and if
+// it is an alias to another variable. For example, this is a string value
+//
+// "valuesByMode": {
+//   "1:0": "Hello"
+// }
+//
+// And this is an alias to another variable:
+//
+// "valuesByMode": {
+//   "1:0": {
+//     "type": "VARIABLE_ALIAS",
+//     "id": "VariableID:1:234"
+//   },
+// }
+//
+#[derive(Deserialize, Serialize, Debug, Clone)]
+#[serde(untagged)]
+pub enum VariableValue {
+    Boolean(bool),
+    Float(f32),
+    String(String),
+    Color(FigmaColor),
+    Alias(VariableAlias),
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct VariableCommon {
+    pub id: String,
+    pub name: String,
+    pub remote: bool,
+    pub key: String,
+    pub variable_collection_id: String,
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")] // Because "resolvedType" is in this format
+#[serde(tag = "resolvedType")] // Maps to different enum values based on "resolvedType"
+pub enum Variable {
```

Comment:
Is this enum having Boolean, Float, String, Color mainly for the resolved type? I mean all of them have 
```
        common: VariableCommon,
        values_by_mode: HashMap<String, VariableValue>,
```
which are the same. So I am wondering why have four enum instead of just `struct Variable` with common, values_by_mode and resolved_type?

Reply:
I also see below in tookit_schema.rs defines Variable again.
So here the definition has to be this way to make the figma json response be able to convert to this Variable and later we can convert this to the toolkit defined Variable? Do I get it right?